### DISCUSSION TITLE
Setting machine specific parameters for Ubuntu live boot

### DISCUSSION THREAD

**Discussion by lucasaka** [ipxe/ipxe#414] [Q&A]
Hi everyone, 
I've been working with setting up an ipxe live boot of Ubuntu to memory. I want to make sure there's a specific config for each machine. I currently have the issue that I cant use cloud-init to set any configuration options. My boot menu is configured as follows:

```
:ubuntu
set ubuntu-version ubuntu-20.04.2.0-desktop-amd64
set ubuntu-folder ${server-address}/ubuntu
set cloud-init ${server-address}/cloud-init/${netX/mac}/
kernel ${ubuntu-folder}/vmlinuz
initrd ${ubuntu-folder}/initrd 
imgargs vmlinuz root=/dev/ram0 ip=dhcp url=${ubuntu-folder}/${ubuntu-version}.iso autoinstall ds=nocloud-net\;s=${cloud-init}
boot
```

Do anyone have any input on what I may have done wrong? I would appreciate any input that helps me move forward with this issue. 

BR

**Comment by NiKiZe** [comment]
What is the actual issue? Any error message?

**Reply by lucasaka** [reply]
No data from the cloud-init folder is instantiated when I reach the desktop. There is no error messages

**Reply by NiKiZe** [reply]
ok so the `ds=nocloud-net\;s=` part is some magic for ubuntu, not sure how much help the iPXE community can be here, have you asked in any ubuntu forum or similar? or even just reading documentation.

We here know a lot about iPXE and network booting, but not about specific distributions internal workings. I don't even know what it's for and not much details was given either.

Where did you find any documentation that or how this should wok in the first place?

**Reply by lucasaka** [reply]
Okey, yea I figured it might be. I've been at this for quite a while but I'll give it a shot somewhere else, thank you though. I havnt really seen documentation however I tried following this example https://forum.ipxe.org/showthread.php?tid=22147 and have been trying different versions of it. 

Thanks

**Reply by NiKiZe** [reply]
First look for documentation from Ubuntu, there should be some official docs
